src/base/faucet/Index.svelte

```diff
@@ -0,0 +1,121 @@
+<script lang="ts">
+  import type { Config } from "@app/config";
+  import { BigNumber } from "@ethersproject/bignumber";
+  import { formatEther, parseUnits } from "@ethersproject/units";
+  import type { ethers } from "ethers";
+  import { onMount } from "svelte";
+  import { navigate } from "svelte-routing";
+  import { getMaxWithdrawAmount, lastWithdrawalByUser, calculateTimeLock } from "./Faucet";
+
+  export let config: Config;
+
+  let amount = 0;
+  let maxWithdrawAmount: BigNumber;
+  let lastWithdrawal: BigNumber;
+  let error: string | undefined;
+
+  async function withdraw() {
+    const [state, message] = await isAbleToWithdraw(amountBN);
+    if (state === true) navigate("/faucet/submit", { state: { amount } });
+    else error = message;
+  }
+
+  async function isAbleToWithdraw(amount: ethers.BigNumber): Promise<[boolean, string?]> {
+    if (amount.isZero()) { return [false, "Not able to withdraw zero tokens"]; }
+    if (parseUnits(amountBN.toString()).gt(maxWithdrawAmount)) return [false, `Reduce amount, max withdrawal is ${formatEther(maxWithdrawAmount)}`];
+    let currentTime = new Date().getTime();
+    let timelock = await calculateTimeLock(amountBN, config);
+    let nextAvailableWithdraw = lastWithdrawal.add(timelock).mul(Math.pow(10,3));
```

Comment:
What's the mul by 1000 for?

Reply:
The timestamp stored in the contract is a 10 digit timestamp based on seconds, while `new Date().getTime()` returns a 13 digit timestamp based on miliseconds.
I can either multiply by 1000 the sum of the lastWIthdrawal and timelock or divide by 1000 the `currentTime`

Reply:
Hmm the other option would be to multiply by 1000 the `block.timestamp` in the faucet contract..

Reply:
Ok. I'd keep the contract as-is, but maybe add a comment explaining what's going on, and make the modification to `currentTime` instead, ie. divide it by 1000.

Reply:
In the end I left the multiplication, since JS is quite bad at figuring it out the difference between a 10 digit and a 13 digit timestamp... And if parsed to a string with some of the `Date` methods it shows a date around January 1970..
I added some comments for better understanding